### PR TITLE
Backend changes for error handling phase 1

### LINKED ISSUE #19019: [Task]: Add missing titles to the Appsmith errors
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

A few Appsmith errors exist that don't have any value in the title field. The goal of this task is to add those missing titles.

### PR DESCRIPTION
**Pull request by subrata71**
(WIP)
This commit adds titles to the AppsmithErrors that don't have any

## Description

- This PR adds missing titles to the generic Appsmith errors
- It captures downstream error messages and code where applicable for each plugin
- It introduces a new convention for defining Appsmith error codes e.g. "PE-GEN-5000", "PE-MYS-5000"

Fixes #19019 #19061 #19063 #19064 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/exceptions/pluginExceptions/AppsmithPluginError.java [hunk 1]
```diff
@@ -5,62 +5,434 @@
 import lombok.Getter;
 
 import java.text.MessageFormat;
+import java.util.regex.Pattern;
 
 @Getter
 public enum AppsmithPluginError {
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/exceptions/pluginExceptions/AppsmithPluginError.java:11]: Could we please break this down into an interface for Appsmith errors and separate enumerated implementations by module? We should implement the newly maintained codes in a way that ensures that plugins error handling is completely encapsulated within the plugin module itself. And similarly for other modules as well.

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/exceptions/pluginExceptions/AppsmithPluginErrorCode.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
+package com.appsmith.external.exceptions.pluginExceptions;
+
+import lombok.Getter;
+
+@Getter
+public enum AppsmithPluginErrorCode {
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/exceptions/pluginExceptions/AppsmithPluginErrorCode.java:6]: Similar to the interface kind of implementation for the other enum.

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/exceptions/pluginExceptions/AppsmithPluginError.java (hunk 1 above)
Comment (subrata71) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/exceptions/pluginExceptions/AppsmithPluginError.java:11]: I am sharing a few (random) facts which are kind of dettering to me.
There are many plugin errors that are not specific to any plugin. These are meant to be common for all. Also, an enum cannot extend another enum. Some error codes are shared by two plugins e.g. GraphQL uses REST API plugin implementation under the hood.

Many of the plugins only have one or two error codes of their own. Do you think just to achieve encapsulation it's worth doing that?

@nidhi-nair

**Comment by subrata71** [comment]
Closing this in favor of this [PR](https://github.com/appsmithorg/appsmith/pull/19755)

**Pull request closed without merging**